daemon/server/handler.go

```diff
@@ -0,0 +1,87 @@
+package server
+
+import (
+	"time"
+
+	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/snapshot"
+)
+
+// Create a new handler that implements daemon protocol and works with domain types.
+func NewHandler(runner runner.Runner, filer snapshot.Filer, pollInterval time.Duration) *Handler {
+	return &Handler{
+		runner:       runner,
+		filer:        filer,
+		pollInterval: pollInterval,
+	}
+}
+
+type Handler struct {
+	runner       runner.Runner
+	filer        snapshot.Filer
+	pollInterval time.Duration
+}
+
+func (h *Handler) CreateSnapshot(path string) (snapshotId string, err error) {
+	return h.filer.Ingest(path)
+}
+
+func (h *Handler) CheckoutSnapshot(snapshotId runner.SnapshotId, dir string) error {
+	// Checkout snapshot in a default directory then move that checkout to the specified dir.
+	co, err := h.filer.Checkout(string(snapshotId))
+	if err == nil {
+		err = co.Disown(dir)
+	}
+	return err
+}
+
+func (h *Handler) Run(cmd *runner.Command) (status runner.ProcessStatus, err error) {
+	// Direct delegation to underlying runner.
+	return h.runner.Run(cmd)
+}
+
+func (h *Handler) Poll(runIds []runner.RunId, timeout time.Duration, returnAll bool) (statuses []runner.ProcessStatus) {
```

Comment:
Should this method (maybe not now, but eventually) just go on Runner? Or be a utility function?

It's been nice when our handlers are very little code other than translating and serializing/deserializing.